routers/private/serv.go

```diff
@@ -157,20 +153,18 @@ func ServCommand(ctx *context.PrivateContext) {
 		results.RepoID = repo.ID
 
 		if repo.IsBeingCreated() {
-			ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
-				"results": results,
-				"type":    "InternalServerError",
-				"err":     "Repository is being created, you could retry after it finished",
+			ctx.JSON(http.StatusInternalServerError, private.ErrServCommand{
+				Results: results,
+				Err:     "Repository is being created, you could retry after it finished",
 			})
 			return
 		}
 
 		// We can shortcut at this point if the repo is a mirror
 		if mode > models.AccessModeRead && repo.IsMirror {
-			ctx.JSON(http.StatusUnauthorized, map[string]interface{}{
-				"results": results,
-				"type":    "ErrMirrorReadOnly",
-				"err":     fmt.Sprintf("Mirror Repository %s/%s is read-only", results.OwnerName, results.RepoName),
+			ctx.JSON(http.StatusUnauthorized, private.ErrServCommand{
+				Results: results,
```

Comment:
Maybe return 400 or 405

Reply:
In the end all different error codes result in `Unauthorized`:
https://github.com/go-gitea/gitea/blob/daa5a2354879a6dff93f431bc7e47670438995ac/cmd/serv.go#L214-L218